spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/chat/model/ToolCallingAutoConfiguration.java

```diff
@@ -48,8 +51,15 @@ public class ToolCallingAutoConfiguration {
 	@Bean
 	@ConditionalOnMissingBean
 	ToolCallbackResolver toolCallbackResolver(GenericApplicationContext applicationContext,
-			List<FunctionCallback> toolCallbacks) {
-		var staticToolCallbackResolver = new StaticToolCallbackResolver(toolCallbacks);
+			ObjectProvider<List<FunctionCallback>> functionCallbacksProvider,
+			ObjectProvider<List<ToolCallback>> toolCallbacksProvider) {
```

Comment:
`ObjectProvider<List<FunctionCallback>>` will provide both FunctionCallback and ToolCallback.
I think we can keep just `ObjectProvider<List<FunctionCallback>>` here, and in the next release we can replace it with `ObjectProvider<List<ToolCallback>>`.

Reply:
> ObjectProvider<List<FunctionCallback>> will provide both FunctionCallback and ToolCallback.
I'm afraid this is not true. Run the `ToolCallingAutoConfigurationTests` and put a breakpoint at `ToolCallingAutoConfiguration` line 59. You will find that the size of the `List<FunctionCallback> allFunctionAndToolCallbacks` is 2, e.g. the number of the total number for the FunctionCallback defined in the test. 
Then the `ObjectProvider<List<ToolCallback>> toolCallbacksProvider` will provide the remaining 3 ToolCallback definitions